lib/archethic/mining.ex

```diff
@@ -36,23 +37,33 @@ defmodule Archethic.Mining do
   @spec start(
           transaction :: Transaction.t(),
           welcome_node_public_key :: Crypto.key(),
-          validation_node_public_keys :: list(Crypto.key())
+          validation_node_public_keys :: list(Crypto.key()),
+          contract_context :: nil | Contract.Context.t()
         ) :: {:ok, pid()}
```

Comment:
contract_context is always passed to the Mining.start function so the default value is useless I think